service/worker/pernamespaceworker_test.go

```diff
@@ -202,9 +202,11 @@ func (s *perNsWorkerManagerSuite) TestMultiplicity() {
 		Enabled: false,
 	}).AnyTimes()
 
-	s.serviceResolver.EXPECT().Lookup("ns3/0").Return(membership.NewHostInfoFromAddress("self"), nil)
-	s.serviceResolver.EXPECT().Lookup("ns3/1").Return(membership.NewHostInfoFromAddress("other"), nil)
-	s.serviceResolver.EXPECT().Lookup("ns3/2").Return(membership.NewHostInfoFromAddress("self"), nil)
+	s.serviceResolver.EXPECT().LookupN("ns3", uint16(3)).Return([]membership.HostInfo{
+		membership.NewHostInfoFromAddress("self"),
+		membership.NewHostInfoFromAddress("other"),
+		membership.NewHostInfoFromAddress("self"),
```

Comment:
I would like to see the mocks in the test not return duplicates, since the real version doesn't, and then test that it starts the right number of workers